qutebrowser/misc/sessions.py

```diff
@@ -142,6 +142,11 @@ def _save_tab(self, tab, active):
                 'url': bytes(item.url().toEncoded()).decode('ascii'),
                 'title': item.title(),
             }
+            if not item_data['title'] and idx == 0: #Most recent first
+                item_data['title'] = tab.title()
+            if not item_data['title']:
+                item_data['title'] = item_data['url']
```

Comment:
I think it'd be better to do this when loading rather than saving - what do you think?